scripts/enforce-version-bump

```diff
@@ -12,34 +12,43 @@ if [[ ${CI_PULL_REQUEST-} ]]; then
   # available on GH under refs/pull/<pr-number>/merge...
   if git fetch --quiet origin refs/pull/$pr_num/merge:refs/remotes/origin/pull-$pr_num/merge; then
     #  ... and then extract the first parent of that hash (the second parent is just our current HEAD commit).
-    base_commit=$(git rev-parse --short origin/pull-$pr_num/merge^1)
-    base_desc="PR base commit $base_commit"
+    base_revision="origin/pull-${pr_num}/merge^1"
   else
     # Fetching refs/pull/<pr-number>/merge sometimes spuriously fails, possibly due to eventual consistency on GitHub.
     # Let's fall back to master in such cases.
-    base_commit=$(git rev-parse --short origin/master)
-    base_desc="master head commit $base_commit"
+    base_revision=origin/master
   fi
-elif [[ ${CIRCLE_BRANCH-} != master ]]; then
-  # For non-PR builds (builds of branches other than master), just compare the version to what's in the current master.
-  base_commit=$(git rev-parse --short origin/master)
-  base_desc="master head commit $base_commit"
-else # master build
+
+elif [[ ${CIRCLE_BRANCH-} == master ]]; then
   exit 0
+
+elif git machete show up &>/dev/null; then
+  base_revision=$(git machete show up 2>/dev/null)
```

Comment:
Note that `git machete show up` (also `update`, `add`) tries to infer the upstream if the current branch isn't managed